### PR TITLE
Add support for render props, and add a few components

### PR DESCRIPTION
**Pull request by bengry**
This PR addresses quite a few things, as the package is not 100% stable just yet.

Main changes to the core package, implemented using the Fabric package are:
- Pass `key` prop as needed, when more than one root child is renderer.
	- Using the index of the child for now, as it seems to be ok. I'm considering changing this to a random GUID or similar, just so there's no chance of issues if the DOM is manually changed (using Angular for example).
- Support [render props](https://reactjs.org/docs/render-props.html) via the following type:
	```typescript
	type InputRendererOptions<TContext extends object> =
		TemplateRef<TContext>
		| ((context: TContext) => HTMLElement)
		| ComponentRef<TContext>
		| RenderComponentOptions<TContext>;
	```
	I believe this type should provide enough flexibility for users, to pass input views as simple or as complex as they'd like.
	- There's currently a limitation on change detection (at least when using `ComponentRef`), which requires the component's change detection to be set to `OnPush`. It might be possible to have child components using the `Default` strategy, but I haven't looked into it yet.
- Add `ReactWrapperComponent<TProps>` as a base component class. This deals with various issues that a lot of components need, as well as provide helper functions for creating and handling render prop inputs (`InputRendererOptions`), or various levels.

Changes to Fabric package include:
* Upgrade to latest `office-ui-fabric-react` version.
* Implement wrapper components for:
- `Breadcrumb` (`fab-breadcrumb`)
- `Button` (all variants)
- `Callout` (`fab-callout`)
- `Checkbox` (`fab-checkbox`)
- `ChoiceGroup` (`fab-choice-group`)
- `ComboBox` (`fab-combo-box`)
- `CommandBar` (`fab-command-bar`)
- `DatePicker` (`fab-date-picker`)
- `Dialog` (`fab-dialog`)
	- Also includes nested components: `DialogFooter`, `DialogContent`)
- `Fabric` (`fab-fabric`)
- `GroupedList` (`fab-grouped-list`)
- `Icon` (`fab-icon`)
- `Image` (`fab-image`)
- `Panel` (`fab-panel`).

Note that I didn't implemented any props marked as `deprecated`. In the hope that maintenance to this package should be minimal. Especially since all the wrappers are manually written.

I tried to stay as close to the render prop names & interfaces as possible (as noted by most types of `@Input()`s that you'll see, such that the official React package docs should be enough to use this package.
I did make minor adjustments as needed to make using this more friendly to Angular components, within reason.

### CODE REVIEW

**Comment by bengry** [comment]
Sorry for messy commit history. I can amend it if needed.

File: apps/demo/src/app/app.component.html [hunk 1]
```diff
@@ -3,21 +3,121 @@ <h1>Angular-React Demo</h1>
   <h2>Getting up and running...</h2>
 
   <ol>
-    <li>Add <i>AngularReactBrowserModule</i>> to <i>app.module.ts</i> in place of the default <i>BrowserModule</i>.</li>
-    <li>Add <i>Fab[component]Module</i> or <i>Mat[component]Module</i> to <i>app.module.ts</i> imports.</li>
+    <li>Add
+      <i>AngularReactBrowserModule</i>> to
```
Comment (benfeely) [inline apps/demo/src/app/app.component.html]: Remove extra ">" after closing i tag.

File: package.json [hunk 2]
```diff
@@ -74,7 +64,7 @@
     "@nrwl/nx": "0.10.0",
     "core-js": "^2.4.1",
     "ngrx-store-freeze": "^0.2.1",
-    "office-ui-fabric-react": "^5.78.0",
+    "office-ui-fabric-react": "^6.20.0",
```
Comment (benfeely) [inline package.json]: Lets lock this to version "6.22.0" until we are able to use the newer version of Typescript.

File: apps/demo/src/app/app.component.html [hunk 3]
```diff
@@ -3,21 +3,121 @@ <h1>Angular-React Demo</h1>
...
+      <i>app.module.ts</i> imports.</li>
     <li>Add Fabric or Material components to your views.</li>
   </ol>
 </div>
 
-<div>
-  <fab-button label="Test Dialog" (onClick)="toggleDialog()"></fab-button>
-  <fab-dialog [hidden]="dialogHidden" (onDismiss)="toggleDialog()" key="d2" title="Fabric [React] Dialog" subText="Use Fabric React components inside your Angular app!" [type]="DialogType.largeHeader">
-    Hello world! {{ counter }}
+<div style="height: 500vh">
```
Comment (benfeely) [inline apps/demo/src/app/app.component.html]: style="..." not needed.

File: apps/demo/src/app/app.component.ts [hunk 4]
```diff
@@ -1,23 +1,218 @@
...
+
+  constructor(injector: Injector, componentFactoryResolver: ComponentFactoryResolver) {
+    this.panelBodyComponent = {
+      componentType: PanelBodyComponent,
+      factoryResolver: componentFactoryResolver,
+      injector: injector
+    };
+  }
+
+  menuProps: IButtonProps['menuProps'] = {
```
Comment (benfeely) [inline apps/demo/src/app/app.component.ts]: Type should be: IContextualMenuProps

File: apps/demo/src/app/app.component.html [hunk 5]
```diff
@@ -3,21 +3,24 @@ <h1>Angular-React Demo</h1>
   <h2>Getting up and running...</h2>
 
   <ol>
-    <li>Add <i>AngularReactBrowserModule</i>> to <i>app.module.ts</i> in place of the default <i>BrowserModule</i>.</li>
-    <li>Add <i>Fab[component]Module</i> or <i>Mat[component]Module</i> to <i>app.module.ts</i> imports.</li>
+    <li>Add
+      <i>AngularReactBrowserModule</i>> to
```
Comment (xjerwa) [inline apps/demo/src/app/app.component.html]: Extra `>` after `</i>`

Comment (bengry) [reply apps/demo/src/app/app.component.html]: Fixed

**Review by benfeely — APPROVED** [review]
Looks good, great leap forward for the library!

**Pull request merged**